core/src/main/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManager.java

```diff
@@ -17,105 +17,34 @@
-				.filter(u -> this.passwordEncoder.matches(presentedPassword, u.getPassword()))
-				.switchIfEmpty(Mono.defer(() -> Mono.error(new BadCredentialsException("Invalid Credentials"))))
-				.flatMap(u -> {
-					boolean upgradeEncoding = this.userDetailsPasswordService != null
-							&& this.passwordEncoder.upgradeEncoding(u.getPassword());
-					if (upgradeEncoding) {
-						String newPassword = this.passwordEncoder.encode(presentedPassword);
-						return this.userDetailsPasswordService.updatePassword(u, newPassword);
-					}
-					return Mono.just(u);
-				})
-				.doOnNext(this.postAuthenticationChecks::check)
-				.map(u -> new UsernamePasswordAuthenticationToken(u, u.getPassword(), u.getAuthorities()) );
-	}
-
-	/**
-	 * The {@link PasswordEncoder} that is used for validating the password. The default is
-	 * {@link PasswordEncoderFactories#createDelegatingPasswordEncoder()}
-	 * @param passwordEncoder the {@link PasswordEncoder} to use. Cannot be null
-	 */
-	public void setPasswordEncoder(PasswordEncoder passwordEncoder) {
-		Assert.notNull(passwordEncoder, "passwordEncoder cannot be null");
-		this.passwordEncoder = passwordEncoder;
+	protected Mono<UserDetails> retrieveUser(String username) {
+		return this.userDetailsService.findByUsername(username);
 	}
 
-	/**
-	 * Sets the {@link Scheduler} used by the {@link UserDetailsRepositoryReactiveAuthenticationManager}.
-	 * The default is {@code Schedulers.parallel()} because modern password encoding is
-	 * a CPU intensive task that is non blocking. This means validation is bounded by the
-	 * number of CPUs. Some applications may want to customize the {@link Scheduler}. For
-	 * example, if users are stuck using the insecure {@link org.springframework.security.crypto.password.NoOpPasswordEncoder}
-	 * they might want to leverage {@code Schedulers.immediate()}.
-	 *
-	 * @param scheduler the {@link Scheduler} to use. Cannot be null.
-	 * @since 5.0.6
-	 */
-	public void setScheduler(Scheduler scheduler) {
-		Assert.notNull(scheduler, "scheduler cannot be null");
-		this.scheduler = scheduler;
-	}
-
-	/**
-	 * Sets the service to use for upgrading passwords on successful authentication.
-	 * @param userDetailsPasswordService the service to use
-	 */
-	public void setUserDetailsPasswordService(
-			ReactiveUserDetailsPasswordService userDetailsPasswordService) {
-		this.userDetailsPasswordService = userDetailsPasswordService;
-	}
-
-	/**
-	 * Sets the strategy which will be used to validate the loaded <tt>UserDetails</tt>
-	 * object after authentication occurs.
-	 *
-	 * @param postAuthenticationChecks The {@link UserDetailsChecker}
-	 * @since 5.2
-	 */
-	public void setPostAuthenticationChecks(UserDetailsChecker postAuthenticationChecks) {
-		Assert.notNull(this.postAuthenticationChecks, "postAuthenticationChecks cannot be null");
-		this.postAuthenticationChecks = postAuthenticationChecks;
+	public void setUserDetailsService(ReactiveUserDetailsService userDetailsService) {
```

Comment:
Is this method necessary? The `ReactiveUserDetailsService userDetailsService ` is a required parameter in the constructor, so it would already be set when an instance of this class is created.